import device classes

jeetugalav opened the pull request:
- [x] API is incorrect/incomplete, correct it
- [x] Config loading is a hack atm, improve it
- [x] Write Tests

Editing your comment to add todo - @agauniyal

Commit:
import device classes

agauniyal commented:
- [ ] No tests are written
- [ ] API is incorrect/incomplete
- [ ] Config loading is bad

agauniyal commented:
The current code is showing wrong coverage too, since the additions are never executed.

agauniyal commented:
Initial API has been corrected, I'll pass another look over it and then push the committed changes.

agauniyal commented:
Initial API has been corrected, I'll go through one more time over it and then push the committed changes.

agauniyal committed:
general maintenance

update .clang-format, add new paths in launch.json, auto-discover
gtest header paths, produce static lib by default with build.sh

agauniyal committed:
upgrade cpp standard requirement to c++14

agauniyal committed:
add debug directory

agauniyal committed:
improve/fix and rewrite device API

agauniyal committed:
lock uws dependency with own fork

agauniyal commented:
Bug found - #16

agauniyal committed:
refactor initial api v2

remove old incorrect api, remove mutexes with either lock guards
or atomic variables, introduce new deviceList api, new types for
deviceList, meta-information fields, necessary virtual functions
fixes #17, fixes #16, fixes #9, fixes #6

agauniyal commented:
Let's write tests later since @jeetugalav isn't contributing atm and this is blocking other issues too.
